client/components/filter-picker/index.js

```diff
@@ -0,0 +1,184 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment, createRef } from '@wordpress/element';
+import { Dropdown, Button, Dashicon } from '@wordpress/components';
+import { stringify as stringifyQueryObject } from 'qs';
+import { omit, find, partial } from 'lodash';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import DropdownButton from 'components/dropdown-button';
+import Link from 'components/link';
+import './style.scss';
+
+class FilterPicker extends Component {
+	constructor( props ) {
+		super( props );
+
+		const { filterPaths, getQueryParamValue } = props;
+		this.state = {
+			nav: filterPaths[ getQueryParamValue() ],
+		};
+		this.listRef = createRef();
+
+		this.getSelectionPath = this.getSelectionPath.bind( this );
+		this.getOtherQueries = this.getOtherQueries.bind( this );
+		this.getSelectedFilter = this.getSelectedFilter.bind( this );
+		this.selectSubFilters = this.selectSubFilters.bind( this );
+		this.getVisibleFilters = this.getVisibleFilters.bind( this );
+		this.goBack = this.goBack.bind( this );
+	}
+
+	getOtherQueries( query ) {
+		const { queryParam } = this.props;
+		return omit( query, queryParam );
+	}
+
+	getSelectionPath( filter ) {
+		const { path, query, queryParam } = this.props;
+		const otherQueries = this.getOtherQueries( query );
+		const data = {
+			[ queryParam ]: filter.value,
+		};
+		const queryString = stringifyQueryObject( Object.assign( otherQueries, data ) );
+		return `${ path }?${ queryString }`;
+	}
+
+	getSelectedFilter() {
+		const { filters, getQueryParamValue, filterPaths } = this.props;
+		const value = getQueryParamValue();
+		const filterPath = filterPaths[ value ];
+		const visibleFilters = this.getVisibleFilters( filters, [ ...filterPath ] );
+		return find( visibleFilters, filter => filter.value === value );
+	}
+
+	getLabels( selectedFilter ) {
+		return [ selectedFilter && selectedFilter.label ];
```

Comment:
If `selectedFilter` is null/falsey couldn't this result in a `[ null ]` being returned here?